src/custom-element.js

```diff
@@ -767,6 +767,8 @@ function createBaseAmpElementProto(win) {
     if (this.isInTemplate_) {
       return;
     }
+    this.implementation_.viewportCallback(false);
+    this.implementation_.unlayoutCallback();
```

Comment:
What about only call `unlayoutCallback` for this.element owned element? Otherwise the component should be responsible for this.